File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 2/12/2026, 3:45:46 PM
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        sub = 0
        seen = []
        if s == " ":
            return 1
        for letter in s:
            if letter in seen:
                sub = max(sub, len(seen))
                index = seen.index(letter)
                seen = seen[index + 1:]
                seen.append(letter)
            else:
                seen.append(letter)
        return max(sub, len(seen))

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 2/12/2026, 3:45:47 PM
#include <set>

class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int start = 0;
        int longest = 0;
        unordered_set<char> seen;
        for (int end = 0; end < s.length(); end++) {
            char currentChar = s[end];
            while (seen.count(currentChar) > 0) {
                seen.erase(s[start]);
                start++;
            }
            seen.insert(currentChar);
            longest = max(longest, end - start + 1);
        }
        return longest;
    }
};